lib/kennel/parts_serializer.rb

```diff
@@ -8,13 +8,9 @@ def initialize(filter:)
 
     def write(parts)
       Progress.progress "Storing" do
-        if filter.tracking_id_filter
-          write_changed(parts)
-        else
-          old = old_paths
-          used = write_changed(parts)
-          (old - used).uniq.each { |p| FileUtils.rm_rf(p) }
-        end
+        old = old_paths
+        used = write_changed(parts)
+        (old - used).uniq.each { |p| FileUtils.rm_rf(p) }
```

Comment:
```suggestion
        File.unlink((old - used).uniq)
```

I think. It might be worth add a code comment / clarifying variable name to point out that `used` is a mixture of files and directories, whereas `old` is only files. In fact, come to think of it: we should make that consistent. I don't think that `used` needs to contain the directories, since we seem to rely on the fact that empty directories disappear via `git`.

Reply:
- old is folders too `Dir["generated/**/*"].grep_v(/json/)` has loads
- git does not remove empty folders so it's nice if we do it

improved the names a bit `old -> existing` and using `FileUtils.rm_rf list`